src/jit/gentree.h

```diff
@@ -574,11 +574,21 @@ struct GenTree
         return isContained() && IsCnsIntOrI() && !isUsedFromSpillTemp();
     }
 
+#if FEATURE_X87_DOUBLES
```

Comment:
IEEE 754 does allow you to evaluate expressions and perform constant folding, provided you follow all the rules set forth in the spec. This would make much of that easier.